src/main/java/com/boraji/tutorial/spring/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,76 @@
+package com.boraji.tutorial.spring.config;
+
+import com.boraji.tutorial.spring.service.impl.UserDetailsServiceImpl;
+import org.apache.log4j.Logger;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+   private final Logger logger = Logger.getLogger(WebSecurityConfig.class);
+
+   @Bean
+   public UserDetailsService userDetailsService() {
+      return new UserDetailsServiceImpl();
+   }
+
+   @Bean
+   public BCryptPasswordEncoder passwordEncoder() {
+      return new BCryptPasswordEncoder();
+   }
+
+   @Override
+   protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+      auth.userDetailsService(userDetailsService()).passwordEncoder(passwordEncoder());
+   }
+
+   @Override
+   protected void configure(HttpSecurity http) throws Exception {
```

Comment:
try to simplify security settings, for example:
for all users, url will start with 'user/'
for admin, url will start with 'admin/'